File: Programmers/challenge1.py
def solution(numbers):
    answer = []

    for i in range(len(numbers)):
        for j in range(i + 1, len(numbers)):
            answer.append(numbers[i] + numbers[j])

    answer = sorted(list(set(answer)))
    return answer

numbers = [2, 1, 3, 4, 1]
# answer = [2,3,4,5,6,7]
print(solution(numbers))

numbers = [5, 0, 2, 7]
# answer = [2,5,7,9,12]
print(solution(numbers))

File: Programmers/challenge1.js
function solution(a, b) {
  var answer = 1234567890;

  answer = a.reduce((total, item, i) => ((total += item * b[i]), total), 0);

  return answer;
}

let a = [1, 2, 3, 4];
let b = [-3, -1, 0, 2];
// answer = 3
console.log(solution(a, b));

a = [-1, 0, 1];
b = [1, 0, -1];
// answer = -2
console.log(solution(a, b));
